components/service/glean/src/main/java/mozilla/components/service/glean/ping/PingMaker.kt

```diff
@@ -44,6 +46,33 @@ internal class PingMaker(
         return timeString.toString()
     }
 
+    /**
+     * Get the ping sequence number for a given ping. This is a
+     * monotonically-increasing value that is updated every time a particular ping
+     * type is sent.
+     *
+     * @param pingName The name of the ping
+     * @return sequence number
+     */
+    private fun getPingSeq(pingName: String): Int {
+        applicationContext.getSharedPreferences(
```

Comment:
Instead of opening the `sharedPreferences` every time, why don't we open them once lazily? Similarly to how it's done [here](https://github.com/mozilla-mobile/android-components/blob/25287691c8b78f7a911436a1de854864e0233e2c/components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericScalarStorageEngine.kt#L46). This way we don't pay the I/O for reading every time we need to increment the sequence numbers.